### PR TITLE
Fix "Discounts not displayed properly in Orders panel" bug #2773

### PR DESCRIPTION
**Pull request by efalayi**
Resolves [#2773](https://github.com/reactioncommerce/reaction/issues/2773)
"Add Discount" is displayed but the actual discount amount is not

**Add Discount** is a tag link which when clicked displays a form or list of discount codes
Total displayed does not reflect discount (if included)

###### Fix
- Register component that displays discount lists/form
```
registerComponent("DiscountList", discountListComponent);
```

- Replace `Add Discount` tag with *discount amount*
```
 {formatPriceString(invoice.discounts)}
```

- Update `getTotalByShop` function to include discount amount
```
const discount = parseFloat(this.getDiscounts());
const discountTotal = Math.max(0, shopSubtotal - discount);
const shopTotal = discountTotal + shopTaxes + shipping;
```

###### Test
```
Login as an admin and set up shipping and payment methods
Add a discount code
Check out and use that code
Go to orders panel
Observe that discount amount is displayed
```

### CODE REVIEW

File: lib/collections/transform/cartOrder.js [hunk 1]
```diff
@@ -203,14 +203,16 @@ export const cartOrderTransform = {
     const subtotals = this.getSubtotalByShop();
     const taxes = this.getTaxesByShop();
     const shipping = parseFloat(this.getShippingTotal());
+    const discount = parseFloat(this.getDiscounts());
```
Comment (zenweasel) [inline lib/collections/transform/cartOrder.js]: I know I said we should be reflecting discount everywhere but this is actually the exception. We haven't finalized how we are handling discounts in a multi-shop scenario and I think this is going to give us odd results, so let's back out this change for now.

**Comment by joykare** [comment]
@efalayi ready for a review?

File: lib/collections/transform/cartOrder.js (hunk 1 above)
Comment (efalayi) [reply lib/collections/transform/cartOrder.js]: @zenweasel when I add a discount, it's not reflected in `Total` displayed on Orders page.

**Comment by zenweasel** [comment]
@efalayi Can you add a little more context to your comment. In a multi-shop scenario or? Where on the orders page using what payment provider? When I tested it in a single shop I was seeing the discount show up

**Comment by efalayi** [comment]
@zenweasel here are screenshots of an order I just made
admin-screen
completed-order

Observe total displayed. They do not match.

**Comment by zenweasel** [comment]
Added this issue #3005 for Stripe, will deal with separately.

**Pull request merged**